src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -81,20 +103,25 @@ interface DocumentMigratorOptions {
 
 interface ActiveMigrations {
   [type: string]: {
-    latestVersion: string;
-    transforms: Array<{
-      version: string;
-      transform: TransformFn;
-    }>;
+    latestVersion?: string;
+    latestReferenceVersion?: string;
+    transforms: Transform[];
   };
 }
 
+interface Transform {
+  version: string;
+  transform: (doc: SavedObjectUnsanitizedDoc) => TransformResult;
```

Comment:
It _could_ be, but I think that makes the implementation harder to grok, because the functions that produce `Transform`s would not use the `options` parameter.

I think what we've run into here is some confusing naming of types, as `TransformFn` predates this new interface.

Perhaps `TransformFn` would be better named as `ApplyTransformsFn`?

Reply:
> Perhaps TransformFn would be better named as ApplyTransformsFn?

++ Yeah let's go with that. Thanks!